jwst/extract_1d/soss_extract/soss_extract.py

```diff
@@ -54,71 +68,62 @@ def get_ref_file_args(ref_files):
         (wavemaps, specprofiles, throughputs, kernels)
     """
     pastasoss_ref = ref_files["pastasoss"]
-    pad = getattr(pastasoss_ref.traces[0], "padding", 0)
-    if pad > 0:
-        do_padding = True
+    specprofile_ref = ref_files["specprofile"]
+    speckernel_ref = ref_files["speckernel"]
+    n_pix = 2 * speckernel_ref.meta.halfwidth + 1
+    speckernel_wv_range = [np.min(speckernel_ref.wavelengths), np.max(speckernel_ref.wavelengths)]
+
+    refmodel_orders = [int(trace.spectral_order) for trace in pastasoss_ref.traces]
+    if orders_requested is None:
+        orders_requested = refmodel_orders
     else:
-        do_padding = False
+        orders_requested = _verify_requested_orders(orders_requested, refmodel_orders)
 
-    (wavemap_o1, wavemap_o2) = _get_soss_wavemaps(
```

Comment:
All uses of the private `_get_soss_wavemaps()` had identical code to handle padding just above the call, and that same padding code is in the public `get_soss_wavemaps`, so it was possible to remove the private one